prog/dftb+/lib_timedep/dynamics.F90

```diff
@@ -3329,9 +3121,18 @@ subroutine getForces(this, movedAccel, totalForce, rho, H1, Sinv, neighbourList,
     end if
 
     if (this%tLaser) then
+      if (.not. this%tdFieldThroughAPI) then
+        this%presentField(:) = this%tdFunction(:, iStep)
+      elseif (.not. this%tdFieldIsSet) then
+        if (iStep == 0) then
+          this%presentField(:) = 0.0_dp
+        else
+          call error("External field has not been set.")
+        end if
+      end if
```

Comment:
This is a duplication of
https://github.com/dftbplus/dftbplus/blob/60123e2f5b9bb891b39dc851a45a8ac60ffe33a0/prog/dftb%2B/lib_timedep/dynamics.F90#L1097-L1105
and would make a good candidate for a small subroutine.